### ISSUE TITLE
Still active?

### ISSUE THREAD

**Issue by callumacrae** [okor/justice#36]
Hey @okor!

I plan on doing some work on this, because we want to use it at @Lostmyname. If I send pull requests, will you deal with them? Just asking because you don't seem to have been active on github for a while :)

Alternatively, would you be willing to add me as a collaborator? I can run you through a few ideas I've had to see if you approve (mostly around ES6 and bower/npm right now).

**Comment by okor** [comment]
Hey @callumacrae 

Justice not dead. The vast majority of my git activity is hidden behind private repos, unfortunately. And I've just been very busy with the development of other (internal to Vox Media) performance related tools. It was also a bit depressing that there were a few issues filed but no pull requests : (

Anyhow, I'm happy to look at PR's, discuss new features/improvements and adding collaborators.

- Adding packaging + publishing would be excellent.
- ES6 is less important IMO but I'm not against it either.
- Custom themes, not really important but kinda cool already has a branch: https://github.com/okor/justice/issues/28
- Memory graphs: I had a branch but maybe never pushed it ... browser support is wacky due to paranoia around revealing memory info in js land ... but possible with flags/start up options.
- Total page size (bytes): would be GREAT but currently no way to do it accurately, would require extra attributes added to performance.getEntries() api ... would really like full HAR access. Might be worth lobbying some Chrome/Firefox developers and bugtrackers.

So those are some ideas. If you want to address one issue of interest at a time that you are interested in, create a separate issue for each. I'll respond to each issue to make sure it's something I think would be good for the project (and prevent you from wasting any time). Once we agree the feature/improvement is legit, then create a PR. I will definitely respond. And if you get really pumped then maybe we could hop on a Google hangout and discuss further collaboration.

**Comment by callumacrae** [comment]
Fair enough, I guess I'm lucky that nearly all of the performance stuff I do is open source (including [this scary graph!](http://lostmyname.github.io/load-time-graph/)).

ES6 is the easiest way to solve the UMD / npm / bower problem—babel can output ES6 modules as UMD modules which would solve #22, #27, and #35.

Total page size would definitely be useful for me, I can look into that one too.

**Comment by okor** [comment]
@callumacrae Great point about ES6. I say go for it. 

Side note: I did a lot of ugly things to make Justice very small (optimized for compiling) but in retrospect it's not that important - since this is an optional thing generally only loaded in dev or for internal employees. And we could make a little optional async loader for folks. I say all of this to mean if the compiled size is significantly increased with the ES6 conversion I think that's OK. And the current structure is actually a pretty tricky to grok as someone else pointed out. So an ES6 update and better organization of code would be better for the project by making it easy to create future pull requests - in addition to making it much easier to package/publish.

**Comment by callumacrae** [comment]
Yeah, my first impression was that it could definitely be easier to contribute (I was thrown by the livereload in examples/simple.html)—perhaps also including livereload directly in the Gruntfile would be a good idea, so typing "grunt" would start a server and open examples/simple.html in the browser.

I've only ever used browser-sync and gulp, so I don't know if what I just said is actually possible, but it's definitely useful if possible.

Working on getting ES6 working with Grunt now :)

**Comment by okor** [comment]
@callumacrae That sounds great. And I didn't spend much time polishing those examples and could definitely be improved. A lot.

**Comment by callumacrae** [comment]
I'm a bit confused, what's happening in jsutice.cache.js? Some of it seems to be settings, but some of it seems to be setting up variables used in other files—is that correct? If so, I'll have to move them into the file they're used because modules.

**Comment by okor** [comment]
@callumacrae Correct. This is a junk drawer, where I stuffed everything that was shared.
